haystack/utils/hf.py

```diff
@@ -39,3 +40,16 @@ def deserialize_hf_model_kwargs(kwargs: Dict[str, Any]):
 
         if isinstance(v, dict):
             deserialize_hf_model_kwargs(v)
+
+
+def resolve_hf_device_map(device_map: Union[str, dict]) -> ComponentDevice:
```

Comment:
@shadeMe I had to add this function since HuggingFace also supports specifying `device_map` as a string.

Reply:
The string here is just for a single device, right? As in `device_map="cuda:0"` or `device_map="cpu"`? `ComponentDevice.from_str` parses the input as a single device. 

Besides, I don't think HF parses string representations of the device map anyway? Correct me if I'm wrong.

Reply:
> The string here is just for a single device, right?

Yes this is correct. I think it was their way of allowing `device_map` also act like `device`. Internally I belive they convert `device_map="cuda:0"` to `device_map={"": "cuda:0"}`